### PR TITLE
rapid reset remediation

### PR DESCRIPTION
**Pull request by everettraven**
**Description of the change:**
Updates dependencies, implementations, testdata, and tests for the default operator-sdk scaffolds and base images for remediation of 
- https://github.com/advisories/GHSA-qppj-fm5r-hxr3
- https://github.com/advisories/GHSA-4374-p667-p6c8

Some notable changes:
- Removal of the go/v3 testdata. This plugin version is deprecated. To keep the testdata and test logic straightforward and easier to maintain, limit the go operator testdata to use go/v4.
- Updating of the deprecated packagemanifests integration tests to use the go/v4 testdata, which required changing the install modes for OLM v0 to only supporting and installing as AllNamespaces mode due to mutating webhooks being present in the testdata.
- Bumps k8s dependencies to v0.28.5
- Bumps Kubebuilder to a commit where the default go/v4 scaffolding includes flags for disabling HTTP/2 on metrics and webhook servers and securely serving the metrics endpoint
- Bumps controller-runtime to v0.16.3 and includes associated changes.
- Adds flags to the helm operator controller for disabling HTTP/2 on metrics and webhook server and securely serving the metrics endpoint

**Motivation for the change:**
- Reduce the manual effort involved after scaffolding a new operator project to resolve:
    - https://github.com/advisories/GHSA-qppj-fm5r-hxr3
    - https://github.com/advisories/GHSA-4374-p667-p6c8

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Comment by acornett21** [comment]
- Relates: #6554

**Comment by acornett21** [comment]
As mentioned in slack the `replace` section of the `go.mod` file can be replaces since all dependency versions there are now past the pinned versions.

**Comment by acornett21** [comment]
I'm not sure if this is planned in another PR, but I think alot of documentation needs to change. I think anything that mentions `option.Namespace` and anything that talks about `cache.MultiNamespacedCacheBuilder`. There could be other things, but those are the main ones I see.

File: hack/generate/samples/internal/go/memcached-with-customization/memcached_with_customization.go [hunk 1]
```diff
@@ -201,82 +186,26 @@ func (mh *Memcached) Run() {
...
-
 // uncommentDefaultKustomizationV4 will uncomment code in config/default/kustomization.yaml
 func (mh *Memcached) uncommentDefaultKustomizationV4() {
 	var err error
 	kustomization := filepath.Join(mh.ctx.Dir, "config", "default", "kustomization.yaml")
 	log.Info("uncommenting config/default/kustomization.yaml to enable webhooks and ca injection")
 
-	err = kbutil.UncommentCode(kustomization, "#- ../webhook", "#")
-	pkg.CheckError("uncomment webhook", err)
+	// err = kbutil.UncommentCode(kustomization, "#- ../webhook", "#")
```
[the remark this answers is not in the document]
Comment (everettraven) [reply hack/generate/samples/internal/go/memcached-with-customization/memcached_with_customization.go]: Should have been removed! Thanks for catching! Updated in https://github.com/operator-framework/operator-sdk/pull/6664/commits/b93b680de41afa45a7be905fc0282b4a03332413

File: internal/cmd/helm-operator/run/cmd.go [hunk 2]
```diff
@@ -279,10 +283,23 @@ func buildNewCacheFunc(watchNamespaces []string, ws []watches.Watch, sch *apimac
...
+		// Don't specify any config values so the defaults specified
+		// a couple lines later are used
+		defaultNamespaces[ns] = cache.Config{}
+		if ns == metav1.NamespaceAll {
+			defaultNamespaces[ns] = cache.Config{
+				LabelSelector: labels.Everything(),
+			}
+		}
+	}
+	fmt.Println("XXX Debug", "defaultNamespaces", defaultNamespaces)
```
[the remark this answers is not in the document]
Comment (everettraven) [reply internal/cmd/helm-operator/run/cmd.go]: Updated in https://github.com/operator-framework/operator-sdk/pull/6664/commits/b93b680de41afa45a7be905fc0282b4a03332413

File: internal/cmd/helm-operator/run/cmd.go [hunk 3]
```diff
@@ -228,7 +228,7 @@ func exitIfUnsupported(options manager.Options) {
 	}
 }
 
-func getWatchNamespaces(namespaces []string) []string {
+func getWatchNamespaces(namespaces map[string]cache.Config) []string {
```
[the remark this answers is not in the document]
Comment (everettraven) [reply internal/cmd/helm-operator/run/cmd.go]: I updated this in https://github.com/operator-framework/operator-sdk/pull/6664/commits/b93b680de41afa45a7be905fc0282b4a03332413 but didn't make it consistent with all the plugins. Going back to do that now

**Comment by everettraven** [comment]
> ot sure if this is planned in another PR, but I think alot of documentation needs to change. I think anything that mentions `option.Namespace` and anything that talks about `cache.MultiNamespacedCacheBuilder`. There could be other things, but those are the main ones I see.

I don't think there is currently any PRs planned to update the docs, but I think it would be good to capture any necessary docs update in a new issue to prevent the size of this PR from blowing up any further. Would you mind capturing the problems you've seen as new issues?

File: internal/cmd/helm-operator/run/cmd.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (everettraven) [reply internal/cmd/helm-operator/run/cmd.go]: Should now be done in https://github.com/operator-framework/operator-sdk/pull/6664/commits/1ce2acd37a80dcab9620be41d5233ce8850d8ef9

**Review by rashmigottipati — APPROVED** [review]
Changes look good to me
/lgtm

**Pull request merged**